### PR TITLE
IBP-2784 Deprecate crop users and persons tables

### PR DESCRIPTION
**Pull request by abatac**

### CODE REVIEW

File: src/main/resources/liquibase/workbench_changelog/12_4_0.xml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		<preConditions onFail="MARK_RAN">
+			<not>
+				<tableExists tableName="crop_persons" />
+			</not>
+		</preConditions>
+		<comment>
+			Create CROP_PERSONS table for storing association between crop and persons
+		</comment>
+		<sql dbms="mysql" splitStatements="true">
+			CREATE TABLE crop_persons (
```
Comment (nahuel-soldevilla) [inline src/main/resources/liquibase/workbench_changelog/12_4_0.xml]: We'll need to remove `users_crop` in the roles branch and change the code to use this table instead if I'm not mistaken. At least with this approach we'll have less conflicts while merging the branches.

cc @darla-leafnode @clarysabel

File: src/main/java/org/generationcp/middleware/manager/WorkbenchDaoFactory.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import org.generationcp.middleware.dao.RoleDAO;
+import org.generationcp.middleware.dao.StandardPresetDAO;
+import org.generationcp.middleware.dao.ToolDAO;
+import org.generationcp.middleware.dao.UserInfoDAO;
+import org.generationcp.middleware.dao.WorkbenchSidebarCategoryDAO;
+import org.generationcp.middleware.dao.WorkbenchSidebarCategoryLinkDAO;
+import org.generationcp.middleware.dao.WorkbenchUserDAO;
+import org.generationcp.middleware.hibernate.HibernateSessionProvider;
+
+public class WorkbenchDaoFactory {
```
Comment (nahuel-soldevilla) [inline src/main/java/org/generationcp/middleware/manager/WorkbenchDaoFactory.java]: We'll need to migrate some methods from the DaoFactory here. I guess it's less confusing to have them here, you can identify quickly on which db you are working on

File: src/main/java/org/generationcp/middleware/pojos/workbench/CropPersonId.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
+package org.generationcp.middleware.pojos.workbench;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.generationcp.middleware.pojos.Person;
+
+import java.io.Serializable;
+
+public class CropPersonId implements Serializable {
```
Comment (nahuel-soldevilla) [inline src/main/java/org/generationcp/middleware/pojos/workbench/CropPersonId.java]: Why not CropPersonDTO?

File: src/main/java/org/generationcp/middleware/service/api/user/UserService.java [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+import org.generationcp.middleware.pojos.workbench.ProjectUserInfo;
+import org.generationcp.middleware.pojos.workbench.Role;
+import org.generationcp.middleware.pojos.workbench.UserInfo;
+import org.generationcp.middleware.pojos.workbench.WorkbenchUser;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public interface UserService {
```
Comment (nahuel-soldevilla) [inline src/main/java/org/generationcp/middleware/service/api/user/UserService.java]: Note to self: manually merge with roles branch service of the same name

File: src/main/java/org/generationcp/middleware/pojos/workbench/CropPersonId.java (hunk 3 above)
Comment (abatac) [reply src/main/java/org/generationcp/middleware/pojos/workbench/CropPersonId.java]: This is actually used as a Hibernate composite primary key class for CropPerson.

**Pull request merged**